File: Leetcode/Python/2938. Separate Black and White Balls/solution.py
# https://leetcode.com/problems/separate-black-and-white-balls/description

class Solution:
    def minimumSteps(self, s: str) -> int:
        swap_count = 0
        put_zero_index = 0

        for idx, val in enumerate(s):
            if val == "0":
                # We can swap only adjacent element, so swap step is (current index - (last zero index + 1))
                swap_count += idx - put_zero_index
                put_zero_index += 1

        return swap_count

File: Leetcode/Typescript/2938. Separate Black and White Balls/solution.ts
// https://leetcode.com/problems/separate-black-and-white-balls/description

function minimumSteps(s: string): number {
  let swapCount = 0,
    zeros = 0;

  for (let idx = 0; idx < s.length; idx++) {
    if (s[idx] === "0") {
      // We can swap only adjacent element, so swap step is (current index - (last zero index + 1))
      swapCount += idx - zeros;
      zeros += 1;
    }
  }

  return swapCount;
}
